torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1464,6 +1464,11 @@ size_t ONNXAssignOutputShape(
     }
   } else if (THPUtils_checkString(output_obj)) {
     // Ignore string, since they are not supported as output in ONNX.
+  } else if (strcmp(THPUtils_typename(output_obj), "NoneType") == 0) {
+    // For cases with tracing, simply ignore NoneType outputs
+    // For cases with scripting, TODO: Add logic to handle NoneType outputs
```

Comment:
For scripting we can cross reference with the output type of this spot, and keep None only if type matches.